rust/processor/src/processors/default_processor.rs

```diff
@@ -293,47 +312,72 @@ impl ProcessorTrait for DefaultProcessor {
         _: Option<u64>,
     ) -> anyhow::Result<ProcessingResult> {
         let processing_start = std::time::Instant::now();
-        let (txns, block_metadata_txns, write_set_changes, wsc_details) =
-            TransactionModel::from_transactions(&transactions);
-
-        let mut block_metadata_transactions = vec![];
-        for block_metadata_txn in block_metadata_txns {
-            block_metadata_transactions.push(block_metadata_txn.clone());
-        }
-        let mut move_modules = vec![];
-        let mut move_resources = vec![];
-        let mut table_items = vec![];
-        let mut current_table_items = AHashMap::new();
-        let mut table_metadata = AHashMap::new();
-        for detail in wsc_details {
-            match detail {
-                WriteSetChangeDetail::Module(module) => move_modules.push(module.clone()),
-                WriteSetChangeDetail::Resource(resource) => move_resources.push(resource.clone()),
-                WriteSetChangeDetail::Table(item, current_item, metadata) => {
-                    table_items.push(item.clone());
-                    current_table_items.insert(
-                        (
-                            current_item.table_handle.clone(),
-                            current_item.key_hash.clone(),
-                        ),
-                        current_item.clone(),
-                    );
-                    if let Some(meta) = metadata {
-                        table_metadata.insert(meta.handle.clone(), meta.clone());
-                    }
-                },
+        let last_transaction_timestamp = transactions.last().unwrap().timestamp.clone();
+        let (
+            txns,
+            block_metadata_transactions,
+            write_set_changes,
+            (move_modules, move_resources, table_items, current_table_items, table_metadata),
+        ) = tokio::task::spawn_blocking(move || {
+            let (txns, block_metadata_txns, write_set_changes, wsc_details) =
```

Comment:
nit: can we move this block to a separate function?

Reply:
ah... this rebase is going to suck 😭